backend/src/Database.hs

```diff
@@ -63,36 +70,65 @@ initializeDatabase = do
           \country TEXT,\
           \PRIMARY KEY(id)\
           \);"
+        execute_
+          conn
+          "CREATE TABLE Ratings (\
+          \id INTEGER NOT NULL,\
+          \playerId INTEGER NOT NULL,\
+          \side TEXT NOT NULL,\
+          \timestamp INTEGER NOT NULL,\
+          \reportId INTEGER NOT NULL,\
+          \ratingBefore INTEGER NOT NULL,\
+          \ratingAfter INTEGER NOT NULL,\
+          \PRIMARY KEY(id),\
+          \FOREIGN KEY(playerId) REFERENCES Players(id),\
+          \FOREIGN KEY(reportId) REFERENCES GameReports(id));"
         execute_ conn "CREATE INDEX idx_reports_timestamp ON GameReports (timestamp DESC);"
-        execute_ conn "CREATE INDEX idx_reports_winner_timestamp ON GameReports (winner, timestamp DESC);"
-        execute_ conn "CREATE INDEX idx_reports_loser_timestamp ON GameReports (loser, timestamp DESC);"
         execute_ conn "CREATE INDEX idx_players_name ON Players (name);"
+        execute_ conn "CREATE INDEX idx_ratings_player ON Ratings (playerId, side, timestamp DESC);"
       _ -> log env.logger "Database already initialized"
 
 getPlayerByName :: SQL.Connection -> PlayerName -> IO (Maybe ReadPlayer)
-getPlayerByName conn name = do
-  result <- query conn "SELECT * FROM Players WHERE name = ?" (Only name)
-  case result of
-    [] -> pure Nothing
-    [player] -> pure (Just player)
-    _ -> throwIO UnexpectedResultException
+getPlayerByName conn name = readZeroOrOne $ query conn "SELECT * FROM Players WHERE name = ?" (Only name)
```

Comment:
Love how much shorter the extra helper function made this

Reply:
I realized a shortcoming in the sqlite-simple library while writing this. Because the return type of `query` is so loose (`[r] for any r`) type inference is pretty useless and the haskell side of things has no choice but to believe whatever you tell it is inside the list. This can manifest in runtime errors if that's not actually what's in the list after the query is performed. 

This is the cost of getting to write very simple untyped SQL statements. Probably worth it for now, but it's got me interested to see what other SQL library options are available that might be more complicated to use, but more type safe

Reply:
Yeahh https://hackage.haskell.org/package/esqueleto